source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -472,6 +469,10 @@ void Filter::onUpstreamEvent(Network::ConnectionEvent event) {
         Upstream::Outlier::Result::SUCCESS);
     onConnectionSuccess();
 
+    getRequestInfo().setRequestedServerName(read_callbacks_->connection().requestedServerName());
+    ENVOY_LOG(debug, "TCP:onData(), requestedServerName: {} ",
```

Comment:
nit: Probably should log "onUpstreamEvent" not "onData" and delete the trailing space in the string.